lib/xilinx/xc7series/ecc.cc

```diff
@@ -36,6 +38,21 @@ uint32_t icap_ecc(uint32_t idx, uint32_t data, uint32_t ecc) {
 	return ecc;
 }
 
+static uint32_t calculateECC(const std::vector<uint32_t>& data) {
+	uint32_t ecc = 0;
+	for (size_t ii = 0; ii < data.size(); ++ii) {
+		ecc = xc7series::icap_ecc(ii, data[ii], ecc);
+	}
+	return ecc;
+}
+
+void updateECC(std::vector<uint32_t>& data) {
+	assert(data.size() != 0);
```

Comment:
Let's actually change this `assert(data.size() >= 0x32)`?
Also lets move 0x32 into a `constexpr size_t`

Reply:
Done